sharkfuser/tests/lit/test_matmul_asm_emitter_noncontiguous.cpp

```diff
@@ -0,0 +1,123 @@
+// TORCH-CHECK:       %result_perm = torch.aten.matmul %arg0_matrix_a_perm, %arg1_matrix_b_perm : !torch.vtensor<[16,64,128],f32>, !torch.vtensor<[16,128,256],f32> -> !torch.vtensor<[16,64,256],f32>
+// TORCH-CHECK:       %permute_C_val_0_matmul_nc = torch.constant.int 0
+// TORCH-CHECK:       %permute_C_val_1_matmul_nc = torch.constant.int 2
+// TORCH-CHECK:       %permute_C_val_2_matmul_nc = torch.constant.int 1
+// TORCH-CHECK:       %permute_C_matmul_nc = torch.prim.ListConstruct %permute_C_val_0_matmul_nc, %permute_C_val_1_matmul_nc, %permute_C_val_2_matmul_nc : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
+// TORCH-CHECK:       %result = torch.aten.permute %result_perm, %permute_C_matmul_nc : !torch.vtensor<[16,64,256],f32>, !torch.list<int> -> !torch.vtensor<[16,256,64],f32>
+// TORCH-CHECK:       torch.overwrite.tensor.contents %result overwrites %result_ : !torch.vtensor<[16,256,64],f32>, !torch.tensor<[16,256,64],f32>
+// TORCH-CHECK:       return
+// TORCH-CHECK:     }
+// TORCH-CHECK:   }
+//
+// LINALG-CHECK:    util.func public @main$async(%[[ARG0:.+]]: !hal.buffer_view, %[[ARG1:.+]]: !hal.buffer_view, %[[ARG2:.+]]: !hal.buffer_view, {{.+}}
+// LINALG-CHECK:      %[[A:.+]] = hal.tensor.import wait(%{{.+}}) => %[[ARG1]] : !hal.buffer_view -> tensor<16x128x64xf32>
+// LINALG-CHECK:      %[[B:.+]] = hal.tensor.import wait(%{{.+}}) => %[[ARG2]] : !hal.buffer_view -> tensor<16x128x256xf32>
+// LINALG-CHECK:      %[[AT:.+]] = linalg.transpose ins(%[[A]] : tensor<16x128x64xf32>) outs(%{{.+}} : tensor<16x64x128xf32>) permutation = [0, 2, 1]
+// LINALG-CHECK:      %[[OUT:.+]] = linalg.batch_matmul ins(%[[AT]], %[[B]] : tensor<16x64x128xf32>, tensor<16x128x256xf32>) outs(%{{.+}} : tensor<16x64x256xf32>) -> tensor<16x64x256xf32>
+// LINALG-CHECK:      %[[OUTT:.+]] = linalg.transpose ins(%[[OUT]] : tensor<16x64x256xf32>) outs(%{{.+}} : tensor<16x256x64xf32>) permutation = [0, 2, 1]
+// LINALG-CHECK:      %{{.+}} = hal.tensor.alias wait(%{{.+}}) => %[[OUTT]] : tensor<16x256x64xf32> to %[[ARG0]] : !hal.buffer_view
+//
+// AMDGPU-STATS-CHECK: "dispatch-count": 1
+// CPU-STATS-CHECK: "dispatch-count": 1
+//
+// clang-format on
+
+#include <fusilli.h>
+
+#include <cstdint>
+#include <iostream>
+#include <memory>
+#include <string>
+
+using namespace fusilli;
+
+static ErrorObject testMatmulAsmEmitterNoncontiguous(const std::string &mode) {
+  int64_t batch = 16, m = 64, k = 128, n = 256;
+  auto graph = std::make_shared<Graph>();
+  graph->setName("matmul_asm_emitter_noncontiguous");
+  graph->setIODataType(DataType::Float).setComputeDataType(DataType::Float);
+
+  // A has non-contiguous layout with last dimension having stride 1
+  auto aT = graph->tensor(TensorAttr()
+                              .setName("arg0_matrix_a")
+                              .setDim({batch, m, k})
+                              .setStride({m * k, 1, m})); // Non-contiguous
```

Comment:
It means that input `a` is transposed. Probably it makes sense to rename test case and file accordingly. What do you think?